main.go

```diff
@@ -143,7 +146,7 @@ func main() {
 	var hook *webhook.WebHook
 	if *enableWebhook {
 		var err error
-		hook, err = webhook.New(kubeClient, crInformerFactory, *webhookCertDir, *webhookSvcNamespace, *webhookSvcName, *webhookPort, *namespace)
+		hook, err = webhook.New(kubeClient, crInformerFactory, *webhookServerCertFile, *webhookServerKeyFile, *webhookCACertFile, *webhookCertReloadInterval, *webhookSvcNamespace, *webhookSvcName, *webhookPort, *namespace)
```

Comment:
Given that we have quite a number of webhook-related config options, I think it makes sense to wrap them all into a struct for example named `WebhookConfig` that has a method to get an instance by parsing flags, e.g., `FromFlags`. And then pass the `WebhookConfig` instance into `New`.

Reply:
Agreed. I have another set of webhook enhancements that I think I'm going to roll into the same PR, so that sounds like the best option.

Reply:
OK, sounds good to me to roll into another PR.